frontend/src/app/trips/[id]/(tabs)/activities.tsx

```diff
@@ -1,47 +1,323 @@
-import { View, ScrollView, TouchableOpacity } from "react-native";
-import React from "react";
+import {
+  View,
+  ScrollView,
+  TouchableOpacity,
+  TextInput,
+  Text,
+} from "react-native";
+import { useState, useEffect } from "react";
 import { useGlobalSearchParams } from "expo-router";
-import { Ionicons } from "@expo/vector-icons";
-import { Dimensions } from "react-native";
+import { Ionicons, Feather } from "@expo/vector-icons";
+import { Dimensions, StyleSheet } from "react-native";
 import ActivityThumbnail from "@/components/ActivityThumbnail";
-import { mockData } from "../../../../mock-data/activities";
 import { ActivityProps } from "@/types";
+import { fetchActivities } from "@/utils/fetchActivities";
+
+type CategoryKeys =
+  | "Dining"
+  | "Entertainment"
+  | "OutdoorRecreation"
+  | "Shopping"
+  | "Services"
+  | "Transportation"
+  | "Wellness";
+
+const categories: Record<CategoryKeys, string[]> = {
+  Dining: [
+    "restaurant",
+    "cafe",
+    "bakery",
+    "bar",
+    "meal_delivery",
+    "meal_takeaway",
+    "food",
+  ],
+  Entertainment: [
+    "movie_theater",
+    "night_club",
+    "amusement_park",
+    "museum",
+    "library",
+    "art_gallery",
+    "bar",
+    "tourist_attraction",
+    "casino",
+    "bowling_alley",
+  ],
+  OutdoorRecreation: [
+    "park",
+    "zoo",
+    "campground",
+    "aquarium",
+    "university",
+    "stadium",
+    "city_hall",
+    "church",
+  ],
+  Shopping: [
+    "clothing_store",
+    "shopping_mall",
+    "book_store",
+    "jewelry_store",
+    "liquor_store",
+    "home_goods_store",
+    "store",
+    "furniture_store",
+    "supermarket",
+    "pet_store",
+    "florist",
+    "convenience_store",
+    "movie_rental",
+    "hardware_store",
+    "",
+  ],
+  Services: [
+    "car_rental",
+    "car_repair",
+    "laundry",
+    "bank",
+    "accounting",
+    "lawyer",
+    "atm",
+    "car_dealer",
+    "plumber",
+    "police",
+    "post_office",
+    "electrician",
+    "electronics_store",
+    "embassy",
+    "fire_station",
+    "storage",
+  ],
+  Transportation: [
+    "airport",
+    "transit_station",
+    "train_station",
+    "subway_station",
+    "bus_station",
+  ],
```

Comment:
Might be a little weird to have transportation as an Activity, but it's a great idea to recommend user the city's available transportation options (Bus, Taxi, Train,...)

Reply:
I commented out that part for now in case we need to work with transportation again while working on the algorithm